src/Symfony/Bundle/SecurityBundle/DependencyInjection/Compiler/AddSessionDomainConstraintPass.php

```diff
@@ -33,16 +33,10 @@ public function process(ContainerBuilder $container)
         $sessionOptions = $container->getParameter('session.storage.options');
         $domainRegexp = empty($sessionOptions['cookie_domain']) ? '%s' : sprintf('(?:%%s|(?:.+\.)?%s)', preg_quote(trim($sessionOptions['cookie_domain'], '.')));
 
-        if ('auto' === ($sessionOptions['cookie_secure'] ?? null)) {
-            $secureDomainRegexp = sprintf('{^https://%s$}i', $domainRegexp);
-            $domainRegexp = 'https?://'.$domainRegexp;
-        } else {
-            $secureDomainRegexp = null;
-            $domainRegexp = (empty($sessionOptions['cookie_secure']) ? 'https?://' : 'https://').$domainRegexp;
-        }
+        $container->setParameter('session.cookie.secure', $sessionOptions['cookie_secure']);
```

Comment:
we should inject the value directly instead of creating a parameter

Reply:
I tried that approach first but it didn’t work, it threw an error about a missing parameter.